cylc/flow/platforms.py

```diff
@@ -489,7 +489,11 @@ def get_all_platforms_for_install_target(install_target):
 def get_random_platform_for_install_target(install_target):
     """Return a randomly selected platform (dict) for given install target."""
     platforms = get_all_platforms_for_install_target(install_target)
-    return random.choice(platforms)
+    try:
+        return random.choice(platforms)
+    except IndexError:
+        # No platforms to choose from
+        return None
```

Comment:
Before this `IndexError` resulted in a strange error the user wouldn't understand, however, now the error will be pushed into the scripts that call the function.

For example in this code we will get a `KeyError`:

https://github.com/cylc/cylc-flow/blob/4947568bb7e247c71e45d722fae495ab643728af/cylc/flow/task_remote_mgr.py#L232-L233

How about raising a `PlatformLookupError`:

```suggestion
        raise PlatformLookupError(
            f'Could not select platform for install target: {install_target}'
        ) from None
```

Reply:
I had changed the code in remote tidy to pick up the None return, to stop the Key Error.
https://github.com/datamel/cylc-flow/blob/96cd9601ad603cb50b4ff3131e6aa225f152e899/cylc/flow/task_remote_mgr.py#L233
Unsure why this updated code is not showing up for you, hmmm.
Happy to raise the error if you would prefer.

Reply:
> Unsure why this updated code is not showing up for you

Because I wasn't looking properly!

Reply:
I think it should be an error because this hides the whole selection stuff so no platform could mean platform, network or configuration issues.

Reply:
Yes, and I suppose for future use of the method, safest to raise error, I'll update that now.